src/conf.c

```diff
@@ -373,7 +373,18 @@ bool config_rename(meshlink_handle_t *mesh, const char *old_conf_subdir, const c
 	snprintf(old_path, sizeof(old_path), "%s" SLASH "%s", mesh->confbase, old_conf_subdir);
 	snprintf(new_path, sizeof(new_path), "%s" SLASH "%s", mesh->confbase, new_conf_subdir);
 
-	return rename(old_path, new_path) == 0;
+	if(rename(old_path, new_path)) {
+		logger(mesh, MESHLINK_ERROR, "Failed to rename `%s' to `%s': %s", old_path, new_path, strerror(errno));
+		return false;
+	}
+
+	if(!sync_path(new_path)) {
```

Comment:
This will not actually do anything useful. The reason is that `sync_path()` will sync the *contents* of that directory to disk, but those have not been changed. It's the name of the directory itself that is changed, which is part of the contents of its parent directory. So if anything, we should call `sync_path(mesh->confbase)` here. Although I don't think it is really necessary; we should already handle the case where power fails before the rename is flushed to disk.